src/main/java/org/jenkins/tools/test/model/hook/HookOrderComparator.java

```diff
@@ -0,0 +1,32 @@
+package org.jenkins.tools.test.model.hook;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+/**
+ * Comparator that will order first based on the {@code @HookOrder}. Objects whose classes are
+ * missing the annotation order will be treated as if they have the default order. Where two objects
+ * have the same order sorting will occur based on the classname.
+ */
+public class HookOrderComparator implements Comparator<Object>, Serializable {
```

Comment:
This seems useful but is cluttering up the diff with an unrelated change. Can we file a separate PR for the hook order feature, with some unit tests, in order to keep this diff smaller and more manageable for the reader?

Reply:
sure

Reply:
#521